### PR TITLE
feat: Add Playwright e2e tests [DEV-4863]

### PR DESCRIPTION
**Pull request by sownak**
Added Playwright tests

### CODE REVIEW

File: packages/credo/src/agent.ts
Comment (Eengineer1) [inline packages/credo/src/agent.ts:1]: What measure do we have against multiple instantiated agents trying to bind a single identical port?

File: docker/docker-compose.yml
Comment (Eengineer1) [inline docker/docker-compose.yml:1]: Here we could use proper service-network isolation as:

```yaml
services:
  mcp-server:
    networks:
      mcp-network:
        aliases:
          - faber
  
  acapy-holder:
    networks:
      mcp-network:
        aliases:
          - holder
  
  acapy-issuer:
    networks:
      mcp-network:
        aliases:
          - issuer

networks:
  mcp-network:
    driver: bridge
```

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -28,6 +28,30 @@ jobs:
       - name: "Run pnpm build"
         run: pnpm build
 
+      - name: Setup hosts file
+        run: |
+          sudo bash -c "echo '127.0.0.1 holder' >> /etc/hosts"
+          sudo bash -c "echo '127.0.0.1 issuer' >> /etc/hosts"
+          sudo bash -c "echo '127.0.0.1 faber' >> /etc/hosts"
```
Comment (Eengineer1) [inline .github/workflows/build.yml:35]: Looks like hardcoded values.

An alternative would be to leverage `jq`:

```suggestion
for service in $(jq -r '.services | keys[]' docker/docker-compose.yml); do
    sudo bash -c "echo '127.0.0.1 $service' >> /etc/hosts"
done
```

File: packages/credo/src/agent.ts
Comment (Eengineer1) [reply packages/credo/src/agent.ts:1]: We need a backoff strategy to address edge cases, e.g. trying periodic increments of the port P as P + 1, in case of port is already bound type of error.

File: packages/server/tests/03-anoncreds.spec.ts
Comment (Eengineer1) [inline packages/server/tests/03-anoncreds.spec.ts:1]: Looks good.

File: packages/server/tests/04-connection.spec.ts
Comment (Eengineer1) [inline packages/server/tests/04-connection.spec.ts:1]: Comprehensive enough. Nice!

File: packages/server/tests/05-credential.spec.ts
Comment (Eengineer1) [inline packages/server/tests/05-credential.spec.ts:1]: Think about:

Adding reusable setup + teardown functions to prepare and flush the test environment properly.

Add it before declaring test cases:

```suggestion
test.beforeAll(async () => {
    await setupServer();
  });

  test.afterAll(async () => {
    await teardownServer();
  });
```

Comment (Eengineer1) [reply packages/server/tests/05-credential.spec.ts:1]: Similarly applied on other test files.

File: packages/server/tests/utils.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+		}
+	}
+	if (!servicesHealthy) {
+		throw new Error('Docker services are not healthy after maximum retries.');
+	}
+	console.log('All services are healthy.');
+}
+
+// Helper function to stop Docker containers
+export function stopDockerServies() {
```
Comment (Eengineer1) [inline packages/server/tests/utils.ts:60]: Typo here.

**Review by Eengineer1 — CHANGES_REQUESTED** [review]

File: packages/credo/src/agent.ts
Comment (sownak) [reply packages/credo/src/agent.ts:1]: Added this in agent.ts

File: .github/workflows/build.yml (hunk 1 above)
Comment (sownak) [reply .github/workflows/build.yml]: jq does not work with yaml. use grep

File: packages/server/tests/05-credential.spec.ts
Comment (sownak) [reply packages/server/tests/05-credential.spec.ts:1]: Added beforeAll and afterAll to all test cases

**Comment by sownak** [comment]
@Eengineer1 All comments addressed.
